mlir/include/mlir/Dialect/Tosa/IR/TosaTypesBase.td

```diff
@@ -145,23 +156,35 @@ class Tosa_TensorOfOrNone<list<Type> allowedTypes, string description = ""> :
 // Tensor types with constrained ranks.
 //===----------------------------------------------------------------------===//
 
-def Tosa_Rank0Tensor : TosaTensorRankOf<[Tosa_AnyNumber], [0]>;
-
+// Scalar tensors: Rank-1 (with only one element)
 def Tosa_ScalarTensor : TosaScalarTensorOf<[Tosa_AnyNumber], [1]>;
+def Tosa_ScalarTensor_Extended : TosaScalarTensorOf<[Tosa_AnyNumber_Extended], [1]>;
```

Comment:
Same here. We need to replicate extended semantics everywhere. Wouldn't it be cleaner to allow all types and restrict on validation passes?

Reply:
Same reason above.

Reply:
removed and updated